From 0b08ec41ff14a3a8c17e3f0ec4d4e1b4bd3944c5 Mon Sep 17 00:00:00 2001
From: Yoni Izutskiver <yizutskiver@habana.ai>
Date: Tue, 31 May 2022 15:05:41 +0300
Subject: [PATCH] [SW-89472] greco: pcie: disable LTR cap offset fix

The configuration to disable LTR support in Greco is incorrect.
To skip LTR in the capability linked list - the "next offset" field
of one capability before LTR, should point to offset 0x1c0,
instead of 0x1f8 as in other projects.

Change-Id: I4e50e8f5851d54fbf4676c3e1d9f0ced583a8958
---

diff --git a/arch/greco/common/common.c b/arch/greco/common/common.c
index 9d895be..d9c705d 100644
--- a/arch/greco/common/common.c
+++ b/arch/greco/common/common.c
@@ -74,7 +74,7 @@
 	regmap_writel(1, mmPSOC_GLOBAL_CONF_UNIT_RST_N);
 }
 
-#define PCIE_CAP_HDR_NXT_OFFSET		0x1f8
+#define PCIE_CAP_HDR_NXT_OFFSET		0x1c0
 #define PCIE_LTR_DISABLE		0
 #define PCIE_EXT_LD_TIMEOUT_US		1152
 
@@ -387,7 +387,7 @@
 void preboot_disable_pcie_ltr(void)
 {
 	u32 cap_hdr_cfg, dev_cap2_reg;
-	/* PCIe LTR configuration, set bits 31:20 to 0x1f8 */
+	/* PCIe LTR configuration, set bits 31:20 to 0x1c0 */
 	cap_hdr_cfg = regmap_readl(mmPCIE_DBI_MARGIN_EXT_CAP_HDR_REG);
 	cap_hdr_cfg &= ~(PCIE_DBI_MARGIN_EXT_CAP_HDR_REG_NEXT_OFFSET_MASK);
 	cap_hdr_cfg |= (PCIE_CAP_HDR_NXT_OFFSET <<
